src/core/ext/filters/client_channel/lb_policy_factory.h

```diff
@@ -66,9 +66,18 @@ typedef struct grpc_lb_addresses {
 grpc_lb_addresses *grpc_lb_addresses_create(
     size_t num_addresses, const grpc_lb_user_data_vtable *user_data_vtable);
 
+/** Deep copy the address from \a src to \a dest. */
+void grpc_lb_address_copy(grpc_lb_address* dest, const grpc_lb_address* src,  
```

Comment:
How about rolling back to not using a separate `grpc_lb_address_copy()`? We can move this function body into the for loop body if that makes the code cleaner.

The reasons are:
- copying `grpc_lb_address` highly depends on `grpc_lb_addresses` information;
- we are going to make copying backends an option of `grpc_lb_addresses_copy()`, so `grpc_lb_address_copy()` won't be reused.